benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,82 @@
+import { isEqual } from 'lodash-es';
+
+export interface BenchmarkValue<T> {
+    /**
+     * The value returned by the benchmarked function
+     */
+    value: T;
+    /**
+     * The average time it took to run the function in ms
+     */
+    time: number;
+}
+
+/**
+ * @param functionToBenchmark the deterministic function that should be benchmarked, it will be run multiple times
+ * @param options additional options for the benchmark, the defaults are {@link defaultOptions}
+ */
+export function benchmark<Value>(
+    functionToBenchmark: () => Value,
+    options: BenchmarkOptions<Value> = {}
+): BenchmarkValue<Value> {
+    // eslint-disable-next-line no-param-reassign
+    options = { ...defaultOptions, ...options };
+    const benchmarkStepValues: BenchmarkValue<Value>[] = [];
+    for (let i = 0; i < options.numberOfIterations!; i++) {
```

Comment:
I'm not sure this is a good idea. Since `numberOfIterations` has type `number | undefined` you can call this function with `options === { numberOfIterations: undefined }` which would crash this, I think.

Reply:
I used `defaults()` instead of the rest operation. Because `defaultOptions` has a default value for `numberOfIterations`, this won't crash anymore.

While the typings are not 100% correct, this should be good enough.